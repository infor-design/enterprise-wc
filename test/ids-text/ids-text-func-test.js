/**
 * @jest-environment jsdom
 */
import IdsText from '../../src/ids-text/ids-text';

describe('IdsText Component', () => {
  let elem;

  beforeEach(async () => {
    const text = new IdsText();
    document.body.appendChild(text);
    elem = document.querySelector('ids-text');
  });

  afterEach(async () => {
    document.body.innerHTML = '';
  });

  it('renders with no errors', () => {
    const errors = jest.spyOn(global.console, 'error');
    elem.remove();
    elem = new IdsText();
    document.body.appendChild(elem);
    expect(document.querySelectorAll('ids-text').length).toEqual(1);
    expect(errors).not.toHaveBeenCalled();
  });

  it('renders correctly', () => {
    expect(elem.outerHTML).toMatchSnapshot();
    elem.fontSize = 24;
    expect(elem.outerHTML).toMatchSnapshot();
    elem.type = 'h1';
    expect(elem.outerHTML).toMatchSnapshot();
  });

  it('renders font size setting', () => {
    elem.fontSize = 24;
    expect(elem.fontSize).toEqual('24');
    expect(document.querySelectorAll('ids-text').length).toEqual(1);
  });

  it('renders font-size setting then removes it', () => {
    elem = new IdsText();
    document.body.appendChild(elem);
    elem.fontSize = 24;
    expect(elem.fontSize).toEqual('24');
    elem.fontSize = null;
    expect(elem.fontSize).toEqual(null);
    expect(elem.getAttribute('font-size')).toEqual(null);
  });

  it('renders font weight setting', () => {
    expect(elem.fontWeight).toEqual('normal');
    elem.fontWeight = 'bold';
    expect(elem.getAttribute('font-weight')).toEqual('bold');

    elem.fontWeight = 'bolder';
    expect(elem.getAttribute('font-weight')).toEqual('bolder');
    elem.fontWeight = undefined;
    expect(elem.fontWeight).toEqual('normal');
  });

  it('renders overflow setting', () => {
    expect(elem.overflow).toEqual('ellipsis');
    elem.overflow = 'none';
    expect(elem.getAttribute('overflow')).toEqual('none');

    elem.overflow = 'ellipsis';
    expect(elem.getAttribute('overflow')).toEqual('ellipsis');
    elem.overflow = undefined;
    expect(elem.overflow).toEqual('ellipsis');
  });

  it('renders type setting', () => {
    elem = new IdsText();
    elem.type = 'h1';
    expect(elem.type).toEqual('h1');
    expect(elem.shadowRoot.querySelectorAll('h1').length).toEqual(1);
  });

  it('renders font-size setting then removes it', () => {
    elem = new IdsText();
    document.body.appendChild(elem);
    elem.type = 'h1';
    expect(elem.type).toEqual('h1');
    elem.type = null;
    expect(elem.type).toEqual(null);
    expect(elem.getAttribute('type')).toEqual(null);
    expect(elem.shadowRoot.querySelectorAll('span').length).toEqual(1);
  });

  it('renders audible setting then removes it', () => { // ids-text audible
    elem = new IdsText();
    document.body.appendChild(elem);
    expect(elem.shadowRoot.querySelectorAll('.audible').length).toEqual(0);
    elem.audible = true;
    expect(elem.shadowRoot.querySelectorAll('.audible').length).toEqual(1);
    elem.audible = false;
    expect(elem.shadowRoot.querySelectorAll('.audible').length).toEqual(0);
  });
});
