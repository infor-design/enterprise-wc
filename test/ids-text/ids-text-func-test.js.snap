// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdsText Component renders correctly 1`] = `"<ids-text></ids-text>"`;

exports[`IdsText Component renders correctly 2`] = `"<ids-text font-size=\\"24\\"></ids-text>"`;

exports[`IdsText Component renders correctly 3`] = `"<ids-text font-size=\\"24\\" type=\\"h1\\"></ids-text>"`;
